### ISSUE TITLE
Level_3 (spec3) NRS_IFU failure with exit status=139

### ISSUE THREAD

**Issue by stscijgbot-jp** [spacetelescope/jwst#6694]
_Issue [JP-2502](https://jira.stsci.edu/browse/JP-2502) was created on JIRA by [Hien Tran](https://jira.stsci.edu/secure/ViewProfile.jspa?name=htran):_

During reprocessing of nops11 and lre5 data with b7.8.2 we saw spec3 error for some NRS_IFU datasets with exit status = 139. An example is [jw00828-o052_20211012t222109_spec3_001](https://pljwdmsweb.stsci.edu/owlgui/logs/directory_view?directory_path=/ifs/archive/ops/jwst/info/owl/logs_old/2021/285/owlmgr_jw00828-o052_20211012t222109_spec3_001_1634078601.234144&fileset_name=jw00828-o052_20211012t222109_spec3_001) (/ifs/archive/ops/jwst/info/owl/logs/owlmgr_jw00828-o052_20211012t222109_spec3_001_1634078601.234144)
```
2021-10-12 23:00:28,456 - stpipe.Spec3Pipeline.cube_build - WARNING - /dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.3.2.20210904-py38/lib/python3.8/site-packages/gwcs/geometry.py:203: RuntimeWarning: invalid value encountered in remainder
lon = np.mod(lon, 360.0 * u.deg if nquant else 360.0)

2021-10-12 23:01:37,908 - stpipe.Spec3Pipeline.cube_build - INFO - Mapping each NIRSpec slice to sky for input file
2021-10-12 23:02:48,439 - stpipe.Spec3Pipeline.cube_build - INFO - Mapping each NIRSpec slice to sky for input file
2021285230359-E-ERROR-/dms/local/jwst/pipeline/pkgs/miniconda3/envs/jwstdp-1.3.2.20210904-py38/lib/python3.8/site-packages/jwst/pipeline/calwebb_spec3.cfg FAILED (exit=139) on jw00828-o052_20211012t222109_spec3_001_asn.json.
2021285230359-I-INFO-restoring SDP python env...
SSB python deactivated; OWL python ready for interactive use.
2021285230359-E-ERROR-exit status=139 
```
The .err log shows 'ValueError: assignment destination is read-only'. Apparently this has also been seen during Dev regression runs of b7.8.2. For example, see {{█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████

**Comment by stscijgbot-jp** [comment]
_Comment by [Catherine Kaleida](https://jira.stsci.edu/secure/ViewProfile.jspa?name=ckaleida) on [JIRA](https://jira.stsci.edu/browse/JP-2502?focusedCommentId=555715#comment-555715):_

Howard to investigate, likely a test pattern data issue.

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-2502?focusedCommentId=567980#comment-567980):_

B7.9 test run 2808 of LRE5-OTB shows that this error is still occurring for jw00828-o052_20220123t010757_spec3_001_asn.json in the calwebb_spec3 cube_build step. Will need to triage to see if a pipeline fix is possible.

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-2502?focusedCommentId=568053#comment-568053):_

Moving this ticket to the JP project, because it's definitely a failure in the Cal pipeline, not due to anything else (other than possibly garbage data).

**Comment by stscijgbot-jp** [comment]
_Comment by [Katie Kaleida](https://jira.stsci.edu/secure/ViewProfile.jspa?name=ckaleida) on [JIRA](https://jira.stsci.edu/browse/JP-2502?focusedCommentId=555715#comment-555715):_

Howard to investigate, likely a test pattern data issue.

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-2502?focusedCommentId=567980#comment-567980):_

B7.9 test run 2808 of LRE5-OTB shows that this error is still occurring for jw00828-o052_20220123t010757_spec3_001_asn.json in the calwebb_spec3 cube_build step. Will need to triage to see if a pipeline fix is possible.

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-2502?focusedCommentId=568053#comment-568053):_

Moving this ticket to the JP project, because it's definitely a failure in the Cal pipeline, not due to anything else (other than possibly garbage data).

**Comment by stscijgbot-jp** [comment]
_Comment by [Jane Morrison](https://jira.stsci.edu/secure/ViewProfile.jspa?name=morrison) on [JIRA](https://jira.stsci.edu/browse/JP-2502?focusedCommentId=569080#comment-569080):_

The error I am getting is:

Segmentation fault: 11

(dev12722) arizonajane3:JP-2502_NRS_IFU morrison$ /Users/morrison/opt/anaconda3/envs/dev12722/lib/python3.9/multiprocessing/resource_tracker.py:216: UserWarning: resource_tracker: There appear to be 2 leaked semaphore objects to clean up at shutdown

  warnings.warn('resource_tracker: There appear to be %d '

So that indicates there is an error in c algorithm side of cube build. The data looks reasonable. After outlier detector there are regions on the slices (for all 5 images)  that are marked with DO_NOT_USE.  It could be that there is no valid data for some of the cube spaxels - do data to combine together memory is not being cleaned up correctly.

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-2502?focusedCommentId=570342#comment-570342):_

Fixed by #6718
